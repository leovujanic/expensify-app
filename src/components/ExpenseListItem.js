import React from 'react';
import connect from "react-redux/es/connect/connect";
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


export const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h2>{description}</h2>
		</Link>
		<p>
			{numeral(amount).format('$0,0.00')} - {moment(createdAt).format('MMMM Do, YYYY')}</p>
	</div>
);

export default connect()(ExpenseListItem);