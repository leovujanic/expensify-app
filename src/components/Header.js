import {NavLink} from "react-router-dom";
import React from "react";
import {connect} from 'react-redux';
import {startLogOut} from "../actions/auth";

export const Header = ({startLogOut}) => (
	<header>
		<h1>Expensify</h1>
		<NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
		<NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
		<NavLink to="/help" activeClassName="is-active">Help</NavLink>
		<button onClick={startLogOut}>Log Out</button>
	</header>
);

const mapDispatchToProps = (dispatch) => ({
	startLogOut: () => dispatch(startLogOut()),
});

export default connect(undefined, mapDispatchToProps)(Header);