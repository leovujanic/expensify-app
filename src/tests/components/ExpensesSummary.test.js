import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';


test('Should correctly render ExpensesSummary with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
	expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary with multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={4000}/>);
	expect(wrapper).toMatchSnapshot();
});