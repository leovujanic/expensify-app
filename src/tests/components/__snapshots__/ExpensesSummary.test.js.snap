// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     of 
    expense
     totalling 
    $235.00
  </h1>
</div>
`;

exports[`Should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     of 
    expenses
     totalling 
    $4,000.00
  </h1>
</div>
`;
